math_opt_benchmark/mst/mst.cc

```diff
@@ -0,0 +1,119 @@
+// Copyright 2020 The MathOpt Benchmark Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "math_opt_benchmark/mst/mst.h"
+
+#include <limits>
+
+#include "absl/strings/str_cat.h"
+#include "glog/logging.h"
+
+namespace math_opt_benchmark {
+
+using ::operations_research::MPConstraint;
+using ::operations_research::MPSolver;
+using ::operations_research::MPVariable;
+
+/**
+ * @param problem_type MPSolver constant specifying which solver to use
+ * @param problem Graph specification with edges, weights, and number of
+ * vertices
+ */
+MSTSolver::MSTSolver(const MPSolver::OptimizationProblemType problem_type,
+                     const MSTProblem &problem)
+    : solver_("MST_solver", problem_type) {
+  solver_.MutableObjective()->SetMinimization();
+  for (int i = 0; i < problem.n; ++i) {
+    x_vars_.push_back(
```

Comment:
We should use a sparse representation here, or it will start getting difficult to scale. You can address this now or in a separate change.